src/time.rs

```diff
@@ -0,0 +1,182 @@
+//! Time structures.
+//!
+//! The `time` modules contains structures used to represent both
+//! absolute and relative time.
+
+use std::{ops, fmt};
+
+static NANOS_PER_SEC: u32 = 1_000_000_000;
+
+/// A representation of an absolute time value.
+#[derive(Debug, Default, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct TimeStamp {
+    secs: u64,
+    nanos: u32,
```

Comment:
Are the ~600 years range of a plain `u64` nanoseconds not enough?

Reply:
My goal here was to implement something similar to `Timespec` and `Duration` from libstd. This is what `Timespec` and `Duration` do, so it is what I used. We certainly could use `TimeStamp(u64)`, but it wasn't that much work to add a seconds and nanoseconds member and most other implementations I found use something like this instead of `TimeStamp(u64)`.